oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/async/ElasticResultRowAsyncIterator.java

```diff
@@ -336,13 +336,10 @@ public void onFailure(Throwable t) {
                     System.currentTimeMillis() - searchStartTime);
             // Check in case errorRef is already set - this seems unlikely since we close the scanner once we hit failure.
             // But still, in case this do happen, we will log a warn.
-            Throwable error = errorRef.getAndSet(t);
+            Throwable error = errorRef.getAndSet(new ElasticAsyncQueryException(t, ElasticIndexUtils.toString(query)));
             if (error != null) {
                 LOG.warn("Error reference for async iterator was previously set to {}. It has now been reset to new error {}", error.getMessage(), t.getMessage());
             }
-
-            LOG.error("Error retrieving data for jcr query [{}] :: Corresponding ES query {} : closing scanner, notifying listeners",
```

Comment:
So we no longer log this, why? Isn't there a risk that we don't log it at all in some cases? Unless I'm mistaken this code was changed recently, because we didn't have sufficient log messages...

Reply:
we were logging it twice with slightly different content. The newly introduced exception shows a message with the entire content. It is printed in `hasNext()`